### PR TITLE
*: Support monthname

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: evaluator/builtin_time.go [hunk 1]
```diff
@@ -192,13 +201,12 @@ func builtinNow(args []types.Datum, _ context.Context) (d types.Datum, err error
...
 	d, err := builtinWeekDay(args, ctx)
 	if err != nil || d.Kind() == types.KindNull {
-		d.SetNull()
 		return d, errors.Trace(err)
 	}
 	weekday := d.GetInt64()
 	if (weekday < 0) || (weekday >= int64(len(mysql.WeekdayNames))) {
 		d.SetNull()
-		return d, errors.Errorf("no name for invalid weekday: %d.", weekday)
+		return d, errors.Trace(errors.Errorf("no name for invalid weekday: %d.", weekday))
```
Comment (coocood) [inline evaluator/builtin_time.go]: `errors.Errorf` do not need to be traced.

File: evaluator/builtin_time.go [hunk 2]
```diff
@@ -423,9 +425,8 @@ func builtinExtract(args []types.Datum, _ context.Context) (d types.Datum, err e
 	}
 
 	if val.Kind() != types.KindMysqlTime {
-		err = errors.Errorf("need time type, but got %T", val)
 		d.SetNull()
-		return d, err
+		return d, errors.Trace(errors.Errorf("need time type, but got %T", val))
```
Comment (coocood) [inline evaluator/builtin_time.go]: Same as above.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**